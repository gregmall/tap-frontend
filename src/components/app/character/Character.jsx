import React from 'react';
import PropTypes from 'prop-types';

const Character = ({ name, image, quote, role }) =>(
  <div>
    <p>{name}</p>
    <img src={image} />
    <p>{quote}</p>
    <p>{role}</p>
  </div>

);
Character.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  quote: PropTypes.string.isRequired,
  role: PropTypes.string.isRequired
};

export default Character;
