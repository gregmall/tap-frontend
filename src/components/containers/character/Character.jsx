import React from 'react';
import PropTypes from 'prop-types';
import styles from './Character.css';

const Character = ({ name, image, role }) =>(
  <div className={styles.Character}>
    <p><h2>{name}</h2></p>
    <img src={image} />
    <p>{role}</p>
  </div>

);
Character.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  role: PropTypes.string.isRequired
};

export default Character;
