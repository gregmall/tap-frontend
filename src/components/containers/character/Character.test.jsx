
import React from 'react';
import { render } from '@testing-library/react';
import Character from './Character';

describe('Character component', () => {
  it('renders Character component', () => {
    const { asFragment } = render(<Character name="Nigel" image="www.rock.com" />);

    expect(asFragment()).toMatchSnapshot();
  });
});
