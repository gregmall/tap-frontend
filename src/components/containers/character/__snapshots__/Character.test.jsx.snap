// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders Character component 1`] = `
<DocumentFragment>
  <div
    class="Character"
  >
    <p>
      Nigel
    </p>
    <img
      src="www.rock.com"
    />
    <p />
  </div>
</DocumentFragment>
`;
